### PR TITLE
React simplemde editor

### PR DESCRIPTION
**Pull request by snorrees**
Introduces a new lib for markdown rendering and configures it to address a bunch of issues:
* it works with next
* the editor can be fully customized as we now expose the full underlying editor api
    * added some examples to the readme to show how this can be done
* image url can be customized after upload 
* removes full screen edit / preview as it does not play well with the studio
* remove debouncing value: this seems to fix the cursor jumping around

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,6 +87,7 @@
     "typescript": "^4.9.4"
   },
   "peerDependencies": {
+    "easymde": ">= 2 < 3",
```
Comment (stipsan) [inline package.json:90]: Isn't it better to do
```suggestion
    "easymde": "^2",
```
To accomplish the same thing?

File: README.md [hunk 2]
```diff
@@ -7,26 +7,20 @@
...
 
 ```
-npm install --save sanity-plugin-markdown
-```
-
-or
-
-```
-yarn add sanity-plugin-markdown
+npm install --save sanity-plugin-markdown easymde
```
Comment (stipsan) [inline README.md:23]: Perhaps set the major version the plugin specifies in its peerDeps as part of the install command?
```suggestion
npm install --save sanity-plugin-markdown easymde@2
```

We don't do this for deps like `react` and `styled-components` as we're testing them daily due to their wide usage in other studio tooling. But for a package like `easymde` maybe it makes sense?

**Review by stipsan — APPROVED** [review]
LGTM 👍

File: package.json (hunk 1 above)
Comment (snorrees) [reply package.json:90]: Thats the semver range used by the react lib, I feel we should mirror it

File: README.md (hunk 2 above)
Comment (snorrees) [reply README.md:23]: We could do this, i guess; but would deviate from react-simplemde which is pulling this

**Review by stipsan — APPROVED** [review]
Still good 👍👍

**Pull request merged**